### ISSUE TITLE
selector-type-no-unknown marks class as Unexpected unknown type selector

### ISSUE THREAD

**Issue by evgeniyefimov** [stylelint/stylelint#2470]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

"selector-type-no-unknown" rule marks class ".progress-bar-component" as "Unexpected unknown type selector"

> Which rule, if any, is this issue related to?

`selector-type-no-unknown`

> What CSS is needed to reproduce this issue?

```
﻿.progress-bar-component {
  display: block;

  .progress {
    background-color: white;
    border: none;
    border-radius: 0;
    box-shadow: 0 0 18px rgba(0, 0, 0, 0.2);
    height: 7px;
  }

  .progress-bar {
    background-color: #7ec242;
  }
}
```

> What stylelint configuration is needed to reproduce this issue?

`stylelint-config-standard`

> Which version of stylelint are you using?

`7.10.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

By stylelint-webpack-plugin.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

"Yes, it's related to SCSS maps."

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors you are getting)?

"The following warnings were flagged: 1:1  ×  Unexpected unknown type selector "﻿"   selector-type-no-unknown"

**Comment by jeddy3** [comment]
@evgeniyefimov Thanks for the report and for using the template.

I'm afraid I can't replicate this.

If the rule was complaining about the `.progress-bar-component` selector the message would read:

```console
1:1 × Unexpected unknown type selector "progress-bar-component" (selector-type-no-unknown)
```

At the moment the quotes have nothing inside of them, which makes me think something else is amiss. Can you help us out by creating a reproducible test repo please, so we can get to the bottom of this?

**Comment by evgeniyefimov** [comment]
@jeddy3 Thanks for quick response.

I've found what the reason is. When I opened this file by brackets, there was a red dot in the begin of file. Guys from brackets' repository said that it was an extra character being sneakily inserted by the copy/paste. Visual Studio and Web Storm are both don't show it [http://prntscr.com/esdftt](url), but Brackets does [http://prntscr.com/esdg48](url) That's why there were empty string in console warning.

Sorry for disturbing you.

**Comment by jeddy3** [comment]
No worries. I'm glad you were able to resolve the problem :)

It looks like that stray character was being parsed as a _part of the selector_ by PostCSS and then, more specifically, as a _type selector within that selector_ by `postcss-selector-parser`, which makes sense.

Closing the issue.
